pkg/kube/client.go

```diff
@@ -1206,3 +1212,37 @@ func SetRevisionForTest(c CLIClient, rev string) CLIClient {
 	tc.revision = rev
 	return tc
 }
+
+func findIstiodMonitoringPort(pod *v1.Pod) int {
+	if v, ok := pod.GetAnnotations()["prometheus.io/port"]; ok {
+		if port, err := strconv.Atoi(v); err == nil {
+			return port
+		}
+	}
+	for _, container := range pod.Spec.Containers {
+		if container.Name != discoveryContainerName {
+			continue
+		}
+		argsStr := strings.Join(container.Args, " ")
+		if !strings.Contains(argsStr, "monitoringAddr") {
```

Comment:
This feels pretty awkward... parsing the args on the container is REALLY coupling us to implementation details.

Reply:
Right, I've revised it to check only the prometheus port, as this will also need to be updated if they change the port.